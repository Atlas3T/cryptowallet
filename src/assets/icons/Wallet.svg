<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 51 51" style="enable-background:new 0 0 51 51;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;stroke:#000000;stroke-width:5;stroke-miterlimit:10;}
	.st1{display:none;}
	.st2{display:inline;fill:none;stroke:#000000;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
	.st3{display:inline;}
</style>
<g id="Wallet">
	<path class="st0" d="M46.1,31.5v5.3c0,2.4-1.9,4.3-4.3,4.3H9.5c-2.4,0-4.3-1.9-4.3-4.3V15.5c0-2.4,1.9-4.3,4.3-4.3h32.3
		c2.4,0,4.3,1.9,4.3,4.3v5.2"/>
	<circle cx="46.1" cy="26.1" r="3.1"/>
</g>
<g id="Payments" class="st1">
	<line class="st2" x1="8" y1="19.8" x2="41.8" y2="19.8"/>
	<polyline class="st2" points="35.8,13.3 42.3,19.8 35.8,26.3 	"/>
	<line class="st2" x1="42.3" y1="32.8" x2="8.5" y2="32.8"/>
	<polyline class="st2" points="14.5,39.3 8,32.8 14.5,26.3 	"/>
</g>
<g id="Exchange" class="st1">
	<path class="st2" d="M15.3,21.7c0-5.6,4.6-10.2,10.2-10.2s10.2,4.6,10.2,10.2"/>
	<polyline class="st2" points="42.2,16.3 35.7,22.8 29.2,16.3 	"/>
	<path class="st2" d="M35.7,30.2c0,5.6-4.6,10.2-10.2,10.2s-10.2-4.6-10.2-10.2"/>
	<polyline class="st2" points="8.8,35.7 15.3,29.2 21.9,35.7 	"/>
</g>
<g id="More" class="st1">
	<circle class="st3" cx="26" cy="15.8" r="7.8"/>
	<circle class="st3" cx="12.3" cy="28.6" r="5"/>
	<circle class="st3" cx="35.3" cy="33.6" r="3.5"/>
</g>
</svg>
