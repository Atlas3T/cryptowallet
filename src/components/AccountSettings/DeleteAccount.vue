<template>
  <q-modal
    v-model="open"
    class="dark-modal"
  >
    <div class="close-button-wrapper">
      <q-btn
        :label="$t('close')"
        color="secondary"
        size="sm"
        @click="closeModal()"
      />
    </div>

    <div>
      <h1 class="over-pinpad-message">
        <span>{{ $t('enterPin') }}</span>
      </h1>

      <PinPad ref="PinPad"/>

      <q-btn
        :label="$t('confirm')"
        color="secondary"
        size="sm"
        @click="confirmPin()"
      />
    </div>
  </q-modal>
</template>

<script>
import { mapState } from 'vuex';
import Account from '@/store/wallet/entities/account';
import PinPad from '@/components/Auth/PinPad';

export default {
  name: 'Pin',
  components: {
    PinPad,
  },
  props: {
    open: {
      type: Boolean,
      required: true,
    },
    pinHash: {
      type: String,
      required: true,
    },
  },
  data() {
    return {
      pin: null,
    };
  },
  computed: {
    ...mapState({
      authenticatedAccount: state => state.settings.authenticatedAccount,
    }),
    account() {
      return this.$store.getters['entities/account/find'](this.authenticatedAccount);
    },
    accounts() {
      return this.$store.getters['entities/account/query']().get();
    },
  },
  created() {
    this.$root.$on('inputPin', (pinArr) => {
      this.pin = pinArr.join('');
    });
  },
  methods: {
    /**
     * Closes the modal and resets state
     */
    closeModal() {
      this.$refs.PinPad.resetState();
      Object.assign(this.$data, this.$options.data.apply(this));
      this.$emit('closePinModal');
    },
    /**
     * Validates PIN and displays delete confirmation dialog
     */
    confirmPin() {
      if (!this.pin) {
        this.$toast.create(10, this.$t('wrongPin'), 500);
        return false;
      }

      if (this.$CWCrypto.bcryptCompareString(this.pin, this.pinHash) === true) {
        this.$q.dialog({
          title: 'Confirm',
          message: `Are you sure you want to delete the account "${this.account.name}"`,
          ok: 'Yes',
          cancel: 'No',
        }).then(() => {
          this.deleteAccount(this.account.id);
        }).catch(() => {
          this.closeModal();
        });
      }

      return false;
    },
    deleteAccount(id) {
      const wasDefault = this.account.default;
      Account.$delete(id);
      this.$store.dispatch('settings/setSelectedAccount', null);

      if (wasDefault && this.accounts.length > 0) {
        Account.$update({
          where: record => record.id === this.accounts[0].id,
          data: { default: true },
        });
      }

      if (this.accounts.length === 0) {
        this.$router.push({ path: '/setup' });
      } else {
        this.$router.push({ path: '/' });
      }

      return false;
    },
  },
};
</script>

<style>
.over-pinpad-message {
  color: white;
  font-size: 1.5rem;
  text-align: center;
  margin-bottom: 1rem;
}
</style>
