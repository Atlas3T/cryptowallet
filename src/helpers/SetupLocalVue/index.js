import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
