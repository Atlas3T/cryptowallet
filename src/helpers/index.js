import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
