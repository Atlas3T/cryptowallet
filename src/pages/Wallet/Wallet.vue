<template>
  <div class="layout">
    <div class="list">
      <WalletsList
        :coins="coins"
        :currency="currency"
      />
    </div>
    <div class="button">
      <AddWallet/>
    </div>
  </div>
</template>

<script>
import WalletsList from '@/components/Wallet/Wallet/WalletsList.vue';
import AddWallet from '@/components/Wallet/Wallet/AddWallet.vue';

export default {
  name: 'Balance',
  components: {
    WalletsList,
    AddWallet,
  },
  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      currency: 'GPB',
      coins: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
