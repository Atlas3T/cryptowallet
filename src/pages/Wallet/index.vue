<template>
  <div class="layout">
    <div class="list">
      <wallet-list
        :wallets="wallets"
        :currency="selectedCurrency"
      />
    </div>
    <div class="button">
      <add-wallet/>
    </div>
  </div>
</template>

<script>
import WalletList from '@/components/Wallet/WalletsList';
import AddWallet from '@/components/Wallet/AddWallet';

export default {
  name: 'Balance',

  components: {
    WalletList,
    AddWallet,
  },

  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      wallets: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },

  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
