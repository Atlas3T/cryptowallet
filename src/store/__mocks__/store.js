import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
