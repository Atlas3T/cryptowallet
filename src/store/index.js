/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
