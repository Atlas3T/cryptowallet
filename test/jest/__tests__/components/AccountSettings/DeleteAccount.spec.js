/* eslint-disable no-magic-numbers */
import { shallowMount } from '@vue/test-utils';
import DeleteAccount from '@/components/AccountSettings/DeleteAccount';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';
import Account from '@/store/wallet/entities/account';
import Wallet from '@/store/wallet/entities/wallet';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Address from '@/store/wallet/entities/address';

const accountData = JSON.parse('{"$id":1,"id":1,"uid":"62b49f90-d795-bea0-7cdf-5c7a7f18d2e3","refresh_token":"U2FsdGVkX1+SrO40Mx8VS2EHhvK2/X3G25XK4+jsH2fgzG9IuCuEHR7ckI57DNe1riwz7lNafU1ecQFezy/Ov1W98Fg2dPHJBV8GGgadEbLsGxOVoe0d8L+05yALcWWaumNnMYIearOs/AMqhZkuwMTM16jtsgyIxgqFYPsXzAqdMI/V1rdh5Wjtkb1ZQQ2s5SBBAkWUFB6yr92ACebzF5TCkRLGlJqBQpAPNEwtMh4TI+rRHFqo9SBDrCoElQGGIq7mhEmhkmv+CKPQX1eD6Q==","salt":"$2a$10$8NJPPf3O6RY3UI98PjOYou","pinHash":"$2a$10$8NJPPf3O6RY3UI98PjOYouYOfeDyVVKD4cPqefFJHmqkhsLwTfEZC","name":"Konrad","locale":"en-gb","currency":"GBP","node":null,"default":true,"seed":"U2FsdGVkX19WR8fHv2QN8AN9s95N/QNMPvTCCas0AUgcsnJGC4El5Y54Y+0R2dNp0cwbaBypbQ/VhPay5WD1cYzIMMZPoknXHEYnvVPbMChRksNc+VyBt6bTriQcDak0bSREJdrqwQX7JDClECKXQWF5w7eMDf1RLgs/cOfSCYI=","wallets":[]}');
const walletData = JSON.parse('{"$id":3,"id":3,"account_id":1,"name":"Bitcoin","displayName":"Bitcoin","symbol":"BTC","sdk":"Bitcoin","network":"BITCOIN_TESTNET","internalChainAddressIndex":0,"externalChainAddressIndex":1,"externalAddress":null,"confirmedBalance":0,"unconfirmedBalance":0,"imported":true,"enabled":true,"lastBlockHeight":0,"parentName":"","parentSdk":"","contractAddress":"","decimals":"","hdWallet":{"bip":49,"ext":{"xpriv":"xprv9zZZrtcYB6bg2sxhoVYYKAM8duZHP1MwV1SYNVLStU6qpY6NCKfuTCeeHe1MsBeYDt7XipqZYtshoGC9ghNZ6EVwugKxUPM7m2tFmExu2mt","xpub":"xpub6DYvGQ9S1U9yFN3AuX5YgJHsBwPmnU5nrEN9Ask4SodphLRWjrz9zzy88v6HFPGJuDoMg7hhEG2yFFhU7Djqx3AoE57hEL748iFp26JLXtC"},"int":{"xpriv":"xprv9zZZrtcYB6bg6LXLr8kweVuqCWUTc5dnFDJ526bEyz5pLHdfHRx2D1UPLE8bZy6hK8f1YpXQHeGF2FWEds5S1MSbTH2boaQRNUoz6vy4MRq","xpub":"xpub6DYvGQ9S1U9yJpboxAHx1drZkYJx1YMdcSDfpUzrYKcoD5xopyGGkonsBXbmCnHteYYMcjgWRU7EyByaGf8Mav4y5BTkYmC2TD2JVPmkJxs"},"type":1,"network":{"name":"BITCOIN_TESTNET","type":"testnet","bip":1,"segwit":true,"discovery":"http://92.207.178.198:3001/api","broadcastUrl":"https://chain.so/api/v2/send_tx/BTCTEST","feeApi":"https://api.blockcypher.com/v1/btc/main","connect":{"messagePrefix":"Bitcoin Signed Message:","bech32":"tb","bip32":{"public":70617039,"private":70615956},"pubKeyHash":111,"scriptHash":196,"wif":239}}},"erc20Wallet":""}');

describe('DeleteAccount component', () => {
  let wrapper;
  let router;
  let storeMocks;

  const defaultProps = {
    open: true,
    pinHash: accountData.pinHash,
  };

  const mockToast = {
    create: jest.fn(),
  };

  const resetStateMock = jest.fn();


  function wrapperInit(options) {
    return shallowMount(DeleteAccount, options);
  }

  function storeInit(custom, propsData, parentComponent = null) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: '/settings' });

    Account.insert({ data: accountData });
    Wallet.insert({ data: walletData });
    Tx.insert({ data: { account_id: 1 } });
    Utxo.insert({ data: { account_id: 1 } });
    Address.insert({ data: { account_id: 1 } });

    wrapper = wrapperInit({
      i18n,
      router,
      localVue,
      propsData,
      parentComponent,
      store: storeMocks.store,
      mocks: {
        errorHandler: jest.fn(),
        $toast: mockToast,
      },
    });
  }

  beforeEach(() => {
    storeInit({}, defaultProps);
    wrapper.vm.$refs.PinPad = {
      resetState: resetStateMock,
    };
  });

  it('renders and matches snapshot', async (done) => {
    setTimeout(() => {
      expect(wrapper.element).toMatchSnapshot();
      done();
    }, 1000);
  });

  describe('methods', () => {
    describe('resetPin()', () => {
      it('resets the pin array', () => {
        wrapper.vm.pin = [0, 1, 2];
        wrapper.vm.resetPin();
        expect(wrapper.vm.pin.length).toBe(0);
      });
    });

    describe('pinInputListener()', () => {
      it('adds an item into the pin array', () => {
        const item = 'item';
        wrapper.vm.resetPin();
        wrapper.vm.pinInputListener(item);
        expect(wrapper.vm.pin[0]).toBe(item);
      });
    });

    describe('attemptUnlock()', () => {
      it('opens the confirmation dialog and resets the PinPad state if valid pin was entered', () => {
        wrapper.vm.pin = [0, 0, 0, 0, 0, 0];
        wrapper.vm.attemptUnlock();

        expect(resetStateMock).toHaveBeenCalled();
        expect(wrapper.vm.pin.length).toBe(0);
        expect(wrapper.vm.confirmDeleteOpen).toBeTruthy();
      });

      it('display an error if an invalid pin was entered', () => {
        wrapper.vm.pin = [1, 2, 3, 4, 5, 6, 7, 8];
        wrapper.vm.attemptUnlock();

        expect(resetStateMock).toHaveBeenCalledTimes(2);
        expect(wrapper.vm.pin.length).toBe(0);
        expect(wrapper.vm.confirmDeleteOpen).toBeFalsy();
        expect(mockToast.create).toHaveBeenCalled();
      });
    });

    describe('closeModal()', () => {
      it('emits closePinModal event and resets state', () => {
        wrapper.vm.pin = [1, 2, 3, 4, 5, 6, 7, 8];
        wrapper.vm.closeModal();

        expect(resetStateMock).toHaveBeenCalledTimes(3);
        expect(wrapper.vm.pin.length).toBe(0);
        expect(wrapper.emitted().closePinModal).toBeTruthy();
      });
    });

    describe('deleteAccount()', () => {
      it('deletes the account and data belonging to it', async (done) => {
        const closeModalMock = jest.fn();
        wrapper.setMethods({
          closeModal: closeModalMock,
        });

        wrapper.vm.deleteAccount();

        setTimeout(() => {
          expect(storeMocks.actions.setLoading.mock.calls[0][1]).toBe(true);
          expect(storeMocks.actions.setLoading.mock.calls[1][1]).toBe(false);
          expect(storeMocks.actions.setLayout.mock.calls[0][1]).toBe('dark');
          expect(storeMocks.actions.setSelectedAccount.mock.calls[0][1]).toBe(null);
          expect(storeMocks.actions.setAuthenticatedAccount.mock.calls[0][1]).toBe(null);
          expect(closeModalMock).toHaveBeenCalled();

          done();
        }, 500);
      });

      it('changes the default account if deleted account was default', async (done) => {
        const closeModalMock = jest.fn();
        wrapper.setMethods({
          closeModal: closeModalMock,
        });

        const accountNotDefault = JSON.parse('{"$id":2,"id":2,"name":"Konrad Not Default","default":false}');
        Account.insert({ data: accountNotDefault });
        expect(Account.all().length).toBe(2);
        wrapper.vm.deleteAccount();

        setTimeout(() => {
          expect(Account.all().length).toBe(1);
          expect(Account.all()[0].id).toBe(2);
          expect(Account.all()[0].default).toBe(true);
          done();
        }, 500);
      });
    });
  });
});
