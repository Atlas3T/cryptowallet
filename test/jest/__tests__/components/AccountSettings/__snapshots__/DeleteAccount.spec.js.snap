// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccount component renders and matches snapshot 1`] = `
<q-dialog-stub
  contentclass="dark-modal"
  maximized="true"
  open="true"
  persistent="true"
  position="standard"
  transitionhide="slide-down"
  transitionshow="slide-up"
>
  <div>
    <div
      class="header-section"
    >
      <div
        class="header-back-button-wrapper"
      >
        <q-btn-stub
          align="center"
          class="icon-btn back-arrow-btn"
          flat="true"
          icon="arrow_back"
          ripple="true"
          size="lg"
        />
      </div>
       
      <h1
        class="header-h1"
      >
        
        Delete Account
      
      </h1>
    </div>
     
    <div
      class="modal-layout-wrapper center"
    >
      <h1
        class="setup with-margin"
      >
        <span>
          
          Enter your PIN
        
        </span>
      </h1>
       
      <pinpad-stub
        mode="delete"
      />
    </div>
     
    <q-dialog-stub
      position="standard"
      transitionhide="scale"
      transitionshow="scale"
    >
      <q-card-stub
        class="dialog"
        style="width: 300px;"
      >
        <q-card-section-stub>
          <h2>
            
            Confirm
          
          </h2>
           
          <p>
            
            Are you sure you want to delete the account Konrad?
          
          </p>
        </q-card-section-stub>
         
        <q-card-actions-stub
          align="right"
        >
          <q-btn-stub
            align="center"
            color="blueish"
            flat="true"
            label="no"
            ripple="true"
          />
           
          <q-btn-stub
            align="center"
            color="blueish"
            flat="true"
            label="yes"
            ripple="true"
          />
        </q-card-actions-stub>
      </q-card-stub>
    </q-dialog-stub>
  </div>
</q-dialog-stub>
`;
